datasquare.go

```diff
@@ -190,34 +177,26 @@ func (ds *dataSquare) ColumnRoots() [][]byte {
 }
 
 func (ds *dataSquare) computeRowProof(x uint, y uint) ([]byte, [][]byte, uint, uint, error) {
-	tree := merkletree.New(ds.hasher)
-	err := tree.SetIndex(uint64(y))
-	if err != nil {
-		return nil, nil, 0, 0, err
-	}
+	tree := ds.treeCreator.New()
 	data := ds.Row(x)
 
 	for i := uint(0); i < ds.width; i++ {
 		tree.Push(data[i])
 	}
 
-	merkleRoot, proof, proofIndex, numLeaves := tree.Prove()
+	merkleRoot, proof, proofIndex, numLeaves := tree.Prove(int(y))
 	return merkleRoot, proof, uint(proofIndex), uint(numLeaves), nil
 }
 
 func (ds *dataSquare) computeColumnProof(x uint, y uint) ([]byte, [][]byte, uint, uint, error) {
-	tree := merkletree.New(ds.hasher)
-	err := tree.SetIndex(uint64(x))
-	if err != nil {
-		return nil, nil, 0, 0, err
-	}
+	tree := ds.treeCreator.New()
 	data := ds.Column(y)
 
 	for i := uint(0); i < ds.width; i++ {
 		tree.Push(data[i])
 	}
-
-	merkleRoot, proof, proofIndex, numLeaves := tree.Prove()
+	// TODO(ismail): check for overflow when casting from uint -> int
+	merkleRoot, proof, proofIndex, numLeaves := tree.Prove(int(x))
 	return merkleRoot, proof, uint(proofIndex), uint(numLeaves), nil
 }
 
```

Comment:
I think the `dataSquare` type is going to need to expose a (some) method(s) for creating a Proof